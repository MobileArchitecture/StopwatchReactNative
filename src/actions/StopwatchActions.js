export const start = () => ({
  type: 'STOPWATCH_START'
})

export const stop = () => ({
  type: 'STOPWATCH_STOP'
})

export const reset = () => ({
  type: 'STOPWATCH_RESET'
})

export const lap = () => ({
  type: 'STOPWATCH_LAP'
})

export const tick = (elapsed) => ({
  type: 'STOPWATCH_TICK',
  elapsed
})
