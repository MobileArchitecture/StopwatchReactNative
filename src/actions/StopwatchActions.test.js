import * as stopwatch from './StopwatchActions'

describe('stopwatch actions', () => {
  it('should be able to trigger start', () => {
    expect(stopwatch.start()).toMatchSnapshot()
  })

  it('should be able to trigger stop', () => {
    expect(stopwatch.stop()).toMatchSnapshot()
  })

  it('should be able to trigger lap', () => {
    expect(stopwatch.lap()).toMatchSnapshot()
  })

  it('should be able to trigger reset', () => {
    expect(stopwatch.reset()).toMatchSnapshot()
  })

  it('should be able to trigger tick', () => {
    expect(stopwatch.tick(125)).toMatchSnapshot()
  })
})
