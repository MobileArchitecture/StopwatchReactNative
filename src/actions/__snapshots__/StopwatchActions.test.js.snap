// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stopwatch actions should be able to trigger lap 1`] = `
Object {
  "type": "STOPWATCH_LAP",
}
`;

exports[`stopwatch actions should be able to trigger reset 1`] = `
Object {
  "type": "STOPWATCH_RESET",
}
`;

exports[`stopwatch actions should be able to trigger start 1`] = `
Object {
  "type": "STOPWATCH_START",
}
`;

exports[`stopwatch actions should be able to trigger stop 1`] = `
Object {
  "type": "STOPWATCH_STOP",
}
`;

exports[`stopwatch actions should be able to trigger tick 1`] = `
Object {
  "elapsed": 125,
  "type": "STOPWATCH_TICK",
}
`;
